### PR TITLE
[SPARK-24268][SQL] Use datatype.catalogString in error messages

### PR DESCRIPTION
**Pull request by mgaido91**
## What changes were proposed in this pull request?

As stated in https://github.com/apache/spark/pull/21321, in the error messages we should use `catalogString`. This is not the case, as SPARK-22893 used `simpleString` in order to have the same representation everywhere and it missed some places.

The PR unifies the messages using alway the `catalogString` representation of the dataTypes in the messages.

## How was this patch tested?

existing/modified UTs

### CODE REVIEW

**Comment by mgaido91** [comment]
cc @gatorsmile @HyukjinKwon

File: sql/catalyst/src/main/scala/org/apache/spark/sql/types/AbstractDataType.scala [hunk 1]
```diff
@@ -145,7 +145,7 @@ abstract class NumericType extends AtomicType {
 }
 
 
-private[sql] object NumericType extends AbstractDataType {
+private[spark] object NumericType extends AbstractDataType {
```
[the remark this answers is not in the document]
Comment (mgaido91) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/types/AbstractDataType.scala]: I am fine changing them all to `private[spark]`. @gatorsmile @HyukjinKwon what do you think?

[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/types/AbstractDataType.scala]: To me, let's leave them out for now, and fix or discuss when the change is actually needed later.

File: mllib/src/main/scala/org/apache/spark/ml/feature/FeatureHasher.scala [hunk 2]
```diff
@@ -208,8 +208,9 @@ class FeatureHasher(@Since("2.3.0") override val uid: String) extends Transforme
       require(dataType.isInstanceOf[NumericType] ||
         dataType.isInstanceOf[StringType] ||
         dataType.isInstanceOf[BooleanType],
-        s"FeatureHasher requires columns to be of NumericType, BooleanType or StringType. " +
-          s"Column $fieldName was $dataType")
+        s"FeatureHasher requires columns to be of ${NumericType.simpleString}, " +
```
Comment (gatorsmile) [inline mllib/src/main/scala/org/apache/spark/ml/feature/FeatureHasher.scala]: Let us change all?

File: mllib/src/main/scala/org/apache/spark/ml/feature/HashingTF.scala [hunk 3]
```diff
@@ -104,7 +104,7 @@ class HashingTF @Since("1.4.0") (@Since("1.4.0") override val uid: String)
   override def transformSchema(schema: StructType): StructType = {
     val inputType = schema($(inputCol)).dataType
     require(inputType.isInstanceOf[ArrayType],
-      s"The input column must be ArrayType, but got $inputType.")
+      s"The input column must be ${ArrayType.simpleString}, but got ${inputType.catalogString}.")
```
Comment (gatorsmile) [inline mllib/src/main/scala/org/apache/spark/ml/feature/HashingTF.scala]: let us change all?

**Comment by gatorsmile** [comment]
LGTM

Thanks! Merged to master.

**Pull request closed without merging**